<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 5 Gene Ontology Analysis | clusterProfiler: universal enrichment tool for functional and comparative study</title>
  <meta name="description" content="Chapter 5 Gene Ontology Analysis | clusterProfiler: universal enrichment tool for functional and comparative study" />
  <meta name="generator" content="bookdown 0.14 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 5 Gene Ontology Analysis | clusterProfiler: universal enrichment tool for functional and comparative study" />
  <meta property="og:type" content="book" />
  
  
  
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 5 Gene Ontology Analysis | clusterProfiler: universal enrichment tool for functional and comparative study" />
  
  
  

<meta name="author" content="Guangchuang Yu" />



  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="chapter4.html"/>
<link rel="next" href="chapter6.html"/>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />









<script src="libs/htmlwidgets-1.5.1/htmlwidgets.js"></script>
<link href="libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="libs/datatables-binding-0.9/datatables.js"></script>
<link href="libs/dt-core-1.10.19/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="libs/dt-core-1.10.19/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="libs/dt-core-1.10.19/js/jquery.dataTables.min.js"></script>
<link href="libs/crosstalk-1.0.0/css/crosstalk.css" rel="stylesheet" />
<script src="libs/crosstalk-1.0.0/js/crosstalk.min.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Preface</a></li>
<li class="chapter" data-level="1" data-path="chapter1.html"><a href="chapter1.html"><i class="fa fa-check"></i><b>1</b> Introduction</a><ul>
<li class="chapter" data-level="1.1" data-path="chapter1.html"><a href="chapter1.html#terminology"><i class="fa fa-check"></i><b>1.1</b> Terminology</a><ul>
<li class="chapter" data-level="1.1.1" data-path="chapter1.html"><a href="chapter1.html#gene-sets-and-pathway"><i class="fa fa-check"></i><b>1.1.1</b> Gene sets and pathway</a></li>
<li class="chapter" data-level="1.1.2" data-path="chapter1.html"><a href="chapter1.html#gene-ontology-go"><i class="fa fa-check"></i><b>1.1.2</b> Gene Ontology (GO)</a></li>
<li class="chapter" data-level="1.1.3" data-path="chapter1.html"><a href="chapter1.html#kyoto-encyclopedia-of-genes-and-genomes-kegg"><i class="fa fa-check"></i><b>1.1.3</b> Kyoto Encyclopedia of Genes and Genomes (KEGG)</a></li>
<li class="chapter" data-level="1.1.4" data-path="chapter1.html"><a href="chapter1.html#other-gene-sets"><i class="fa fa-check"></i><b>1.1.4</b> Other gene sets</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="2" data-path="chapter2.html"><a href="chapter2.html"><i class="fa fa-check"></i><b>2</b> Functional Enrichment Analysis Methods</a><ul>
<li class="chapter" data-level="2.1" data-path="chapter2.html"><a href="chapter2.html#over-representation-analysis"><i class="fa fa-check"></i><b>2.1</b> Over Representation Analysis</a></li>
<li class="chapter" data-level="2.2" data-path="chapter2.html"><a href="chapter2.html#gene-set-enrichment-analysis"><i class="fa fa-check"></i><b>2.2</b> Gene Set Enrichment Analysis</a></li>
<li class="chapter" data-level="2.3" data-path="chapter2.html"><a href="chapter2.html#leading-edge-analysis-and-core-enriched-genes"><i class="fa fa-check"></i><b>2.3</b> Leading edge analysis and core enriched genes</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="chapter3.html"><a href="chapter3.html"><i class="fa fa-check"></i><b>3</b> Universal enrichment analysis</a><ul>
<li class="chapter" data-level="3.1" data-path="chapter3.html"><a href="chapter3.html#input-data"><i class="fa fa-check"></i><b>3.1</b> Input data</a></li>
<li class="chapter" data-level="3.2" data-path="chapter3.html"><a href="chapter3.html#wikipathways-analysis"><i class="fa fa-check"></i><b>3.2</b> WikiPathways analysis</a></li>
<li class="chapter" data-level="3.3" data-path="chapter3.html"><a href="chapter3.html#cell-marker"><i class="fa fa-check"></i><b>3.3</b> Cell Marker</a></li>
<li class="chapter" data-level="3.4" data-path="chapter3.html"><a href="chapter3.html#msigdb-analysis"><i class="fa fa-check"></i><b>3.4</b> MSigDb analysis</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="chapter4.html"><a href="chapter4.html"><i class="fa fa-check"></i><b>4</b> Disease analysis</a><ul>
<li class="chapter" data-level="4.1" data-path="chapter4.html"><a href="chapter4.html#enrichdo-function"><i class="fa fa-check"></i><b>4.1</b> <code>enrichDO</code> function</a></li>
<li class="chapter" data-level="4.2" data-path="chapter4.html"><a href="chapter4.html#enrichncg-function"><i class="fa fa-check"></i><b>4.2</b> <code>enrichNCG</code> function</a></li>
<li class="chapter" data-level="4.3" data-path="chapter4.html"><a href="chapter4.html#enrichdgn-and-enrichdgnv-functions"><i class="fa fa-check"></i><b>4.3</b> <code>enrichDGN</code> and <code>enrichDGNv</code> functions</a></li>
<li class="chapter" data-level="4.4" data-path="chapter4.html"><a href="chapter4.html#gsedo-fuction"><i class="fa fa-check"></i><b>4.4</b> <code>gseDO</code> fuction</a></li>
<li class="chapter" data-level="4.5" data-path="chapter4.html"><a href="chapter4.html#gsencg-fuction"><i class="fa fa-check"></i><b>4.5</b> <code>gseNCG</code> fuction</a></li>
<li class="chapter" data-level="4.6" data-path="chapter4.html"><a href="chapter4.html#gsedgn-fuction"><i class="fa fa-check"></i><b>4.6</b> <code>gseDGN</code> fuction</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="chapter5.html"><a href="chapter5.html"><i class="fa fa-check"></i><b>5</b> Gene Ontology Analysis</a><ul>
<li class="chapter" data-level="5.1" data-path="chapter5.html"><a href="chapter5.html#supported-organisms"><i class="fa fa-check"></i><b>5.1</b> Supported organisms</a></li>
<li class="chapter" data-level="5.2" data-path="chapter5.html"><a href="chapter5.html#go-classification"><i class="fa fa-check"></i><b>5.2</b> GO classification</a></li>
<li class="chapter" data-level="5.3" data-path="chapter5.html"><a href="chapter5.html#go-over-representation-test"><i class="fa fa-check"></i><b>5.3</b> GO over-representation test</a><ul>
<li class="chapter" data-level="5.3.1" data-path="chapter5.html"><a href="chapter5.html#drop-specific-go-terms-or-level"><i class="fa fa-check"></i><b>5.3.1</b> drop specific GO terms or level</a></li>
<li class="chapter" data-level="5.3.2" data-path="chapter5.html"><a href="chapter5.html#test-go-at-sepcific-level"><i class="fa fa-check"></i><b>5.3.2</b> test GO at sepcific level</a></li>
<li class="chapter" data-level="5.3.3" data-path="chapter5.html"><a href="chapter5.html#reduce-redundancy-of-enriched-go-terms"><i class="fa fa-check"></i><b>5.3.3</b> reduce redundancy of enriched GO terms</a></li>
</ul></li>
<li class="chapter" data-level="5.4" data-path="chapter5.html"><a href="chapter5.html#go-gene-set-enrichment-analysis"><i class="fa fa-check"></i><b>5.4</b> GO Gene Set Enrichment Analysis</a></li>
<li class="chapter" data-level="5.5" data-path="chapter5.html"><a href="chapter5.html#go-semantic-similarity-analysis"><i class="fa fa-check"></i><b>5.5</b> GO Semantic Similarity Analysis</a><ul>
<li class="chapter" data-level="5.5.1" data-path="chapter5.html"><a href="chapter5.html#go-analysis-for-non-model-organisms"><i class="fa fa-check"></i><b>5.5.1</b> GO analysis for non-model organisms</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="6" data-path="chapter6.html"><a href="chapter6.html"><i class="fa fa-check"></i><b>6</b> KEGG analysis</a><ul>
<li class="chapter" data-level="6.1" data-path="chapter6.html"><a href="chapter6.html#kegg-over-representation-test"><i class="fa fa-check"></i><b>6.1</b> KEGG over-representation test</a></li>
<li class="chapter" data-level="6.2" data-path="chapter6.html"><a href="chapter6.html#kegg-gene-set-enrichment-analysis"><i class="fa fa-check"></i><b>6.2</b> KEGG Gene Set Enrichment Analysis</a></li>
<li class="chapter" data-level="6.3" data-path="chapter6.html"><a href="chapter6.html#kegg-module-over-representation-test"><i class="fa fa-check"></i><b>6.3</b> KEGG Module over-representation test</a></li>
<li class="chapter" data-level="6.4" data-path="chapter6.html"><a href="chapter6.html#kegg-module-gene-set-enrichment-analysis"><i class="fa fa-check"></i><b>6.4</b> KEGG Module Gene Set Enrichment Analysis</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="chapter7.html"><a href="chapter7.html"><i class="fa fa-check"></i><b>7</b> MSigDb analysis</a></li>
<li class="chapter" data-level="8" data-path="chapter8.html"><a href="chapter8.html"><i class="fa fa-check"></i><b>8</b> Reactome pathway analysis</a></li>
<li class="chapter" data-level="9" data-path="chapter9.html"><a href="chapter9.html"><i class="fa fa-check"></i><b>9</b> MeSH Enrichment Analysis</a></li>
<li class="chapter" data-level="10" data-path="chapter10.html"><a href="chapter10.html"><i class="fa fa-check"></i><b>10</b> Functional enrichment analysis of genomic coordinations</a></li>
<li class="chapter" data-level="11" data-path="chapter11.html"><a href="chapter11.html"><i class="fa fa-check"></i><b>11</b> Biological theme comparison</a><ul>
<li class="chapter" data-level="11.1" data-path="chapter11.html"><a href="chapter11.html#formula-interface-of-comparecluster"><i class="fa fa-check"></i><b>11.1</b> Formula interface of compareCluster</a></li>
<li class="chapter" data-level="11.2" data-path="chapter11.html"><a href="chapter11.html#visualization-of-profile-comparison"><i class="fa fa-check"></i><b>11.2</b> Visualization of profile comparison</a></li>
</ul></li>
<li class="chapter" data-level="12" data-path="chapter12.html"><a href="chapter12.html"><i class="fa fa-check"></i><b>12</b> Visualization of Functional Enrichment Result</a><ul>
<li class="chapter" data-level="12.1" data-path="chapter12.html"><a href="chapter12.html#bar-plot"><i class="fa fa-check"></i><b>12.1</b> Bar Plot</a></li>
<li class="chapter" data-level="12.2" data-path="chapter12.html"><a href="chapter12.html#dot-plot"><i class="fa fa-check"></i><b>12.2</b> Dot plot</a></li>
<li class="chapter" data-level="12.3" data-path="chapter12.html"><a href="chapter12.html#gene-concept-network"><i class="fa fa-check"></i><b>12.3</b> Gene-Concept Network</a></li>
<li class="chapter" data-level="12.4" data-path="chapter12.html"><a href="chapter12.html#heatmap-like-functional-classification"><i class="fa fa-check"></i><b>12.4</b> Heatmap-like functional classification</a></li>
<li class="chapter" data-level="12.5" data-path="chapter12.html"><a href="chapter12.html#enrichment-map"><i class="fa fa-check"></i><b>12.5</b> Enrichment Map</a></li>
<li class="chapter" data-level="12.6" data-path="chapter12.html"><a href="chapter12.html#upset-plot"><i class="fa fa-check"></i><b>12.6</b> UpSet Plot</a></li>
<li class="chapter" data-level="12.7" data-path="chapter12.html"><a href="chapter12.html#ridgeline-plot-for-expression-distribution-of-gsea-result"><i class="fa fa-check"></i><b>12.7</b> ridgeline plot for expression distribution of GSEA result</a></li>
<li class="chapter" data-level="12.8" data-path="chapter12.html"><a href="chapter12.html#running-score-and-preranked-list-of-gsea-result"><i class="fa fa-check"></i><b>12.8</b> running score and preranked list of GSEA result</a></li>
<li class="chapter" data-level="12.9" data-path="chapter12.html"><a href="chapter12.html#pubmed-trend-of-enriched-terms"><i class="fa fa-check"></i><b>12.9</b> pubmed trend of enriched terms</a></li>
<li class="chapter" data-level="12.10" data-path="chapter12.html"><a href="chapter12.html#goplot"><i class="fa fa-check"></i><b>12.10</b> goplot</a></li>
<li class="chapter" data-level="12.11" data-path="chapter12.html"><a href="chapter12.html#browsekegg"><i class="fa fa-check"></i><b>12.11</b> browseKEGG</a></li>
<li class="chapter" data-level="12.12" data-path="chapter12.html"><a href="chapter12.html#pathview-from-pathview-package"><i class="fa fa-check"></i><b>12.12</b> pathview from pathview package</a></li>
</ul></li>
<li class="chapter" data-level="13" data-path="chapter13.html"><a href="chapter13.html"><i class="fa fa-check"></i><b>13</b> dplyr verbs for clusterProfiler</a><ul>
<li class="chapter" data-level="13.1" data-path="chapter13.html"><a href="chapter13.html#filter"><i class="fa fa-check"></i><b>13.1</b> filter</a></li>
<li class="chapter" data-level="13.2" data-path="chapter13.html"><a href="chapter13.html#arrange"><i class="fa fa-check"></i><b>13.2</b> arrange</a></li>
<li class="chapter" data-level="13.3" data-path="chapter13.html"><a href="chapter13.html#select"><i class="fa fa-check"></i><b>13.3</b> select</a></li>
<li class="chapter" data-level="13.4" data-path="chapter13.html"><a href="chapter13.html#mutate"><i class="fa fa-check"></i><b>13.4</b> mutate</a></li>
<li class="chapter" data-level="13.5" data-path="chapter13.html"><a href="chapter13.html#slice"><i class="fa fa-check"></i><b>13.5</b> slice</a></li>
<li class="chapter" data-level="13.6" data-path="chapter13.html"><a href="chapter13.html#summarise"><i class="fa fa-check"></i><b>13.6</b> summarise</a></li>
</ul></li>
<li class="chapter" data-level="14" data-path="chapter14.html"><a href="chapter14.html"><i class="fa fa-check"></i><b>14</b> Useful utilities</a><ul>
<li class="chapter" data-level="14.1" data-path="chapter14.html"><a href="chapter14.html#bitr"><i class="fa fa-check"></i><b>14.1</b> <code>bitr</code>: Biological Id TranslatoR</a><ul>
<li class="chapter" data-level="14.1.1" data-path="chapter14.html"><a href="chapter14.html#bitr_kegg"><i class="fa fa-check"></i><b>14.1.1</b> <code>bitr_kegg</code>: converting biological IDs using KEGG API</a></li>
</ul></li>
<li class="chapter" data-level="14.2" data-path="chapter14.html"><a href="chapter14.html#setReadable"><i class="fa fa-check"></i><b>14.2</b> <code>setReadable</code>: translating gene IDs to human readable symbols</a></li>
</ul></li>
<li class="appendix"><span><b>Appendix</b></span></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i>References</a></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">clusterProfiler: universal enrichment tool for functional and comparative study</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="chapter5" class="section level1">
<h1><span class="header-section-number">Chapter 5</span> Gene Ontology Analysis</h1>
<div id="supported-organisms" class="section level2">
<h2><span class="header-section-number">5.1</span> Supported organisms</h2>
<p>GO analyses (<code>groupGO()</code>, <code>enrichGO()</code> and <code>gseGO()</code>) support organisms that have an <code>OrgDb</code> object available.</p>
<p>Bioconductor have already provide <code>OrgDb</code> for about <a href="http://bioconductor.org/packages/release/BiocViews.html#___OrgDb">20 species</a>. User can query <code>OrgDb</code> online by <a href="https://www.bioconductor.org/packages/AnnotationHub">AnnotationHub</a> or build their own by <a href="https://www.bioconductor.org/packages/AnnotationForge">AnnotationForge</a>. An example can be found in the <a href="https://bioconductor.org/packages/devel/bioc/vignettes/GOSemSim/inst/doc/GOSemSim.html#supported-organisms">vignette</a> of <a href="https://www.bioconductor.org/packages/GOSemSim">GOSemSim</a>.</p>
<p>If user have GO annotation data (in data.frame format with first column of gene ID and second column of GO ID), they can use <code>enricher()</code> and <code>gseGO()</code> functions to perform over-representation test and gene set enrichment analysis.</p>
<p>If genes are annotated by direction annotation, it should also annotated by its ancestor GO nodes (indirect annation). If user only has direct annotation, they can pass their annotation to <code>buildGOmap</code> function, which will infer indirection annotation and generate a <code>data.frame</code> that suitable for both <code>enricher()</code> and <code>gseGO()</code>.</p>
</div>
<div id="go-classification" class="section level2">
<h2><span class="header-section-number">5.2</span> GO classification</h2>
<p>In <a href="https://www.bioconductor.org/packages/clusterProfiler">clusterProfiler</a>, <code>groupGO</code> is designed for gene classification based on GO distribution at a specific level. Here we use dataset <code>geneList</code> provided by <a href="https://www.bioconductor.org/packages/DOSE">DOSE</a>. Please refer to vignette of <a href="https://www.bioconductor.org/packages/DOSE">DOSE</a> for more details.</p>
<div class="sourceCode" id="cb53"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb53-1"><a href="chapter5.html#cb53-1"></a><span class="kw">library</span>(clusterProfiler)</span>
<span id="cb53-2"><a href="chapter5.html#cb53-2"></a><span class="kw">data</span>(geneList, <span class="dt">package=</span><span class="st">&quot;DOSE&quot;</span>)</span>
<span id="cb53-3"><a href="chapter5.html#cb53-3"></a>gene &lt;-<span class="st"> </span><span class="kw">names</span>(geneList)[<span class="kw">abs</span>(geneList) <span class="op">&gt;</span><span class="st"> </span><span class="dv">2</span>]</span>
<span id="cb53-4"><a href="chapter5.html#cb53-4"></a>gene.df &lt;-<span class="st"> </span><span class="kw">bitr</span>(gene, <span class="dt">fromType =</span> <span class="st">&quot;ENTREZID&quot;</span>,</span>
<span id="cb53-5"><a href="chapter5.html#cb53-5"></a>        <span class="dt">toType =</span> <span class="kw">c</span>(<span class="st">&quot;ENSEMBL&quot;</span>, <span class="st">&quot;SYMBOL&quot;</span>),</span>
<span id="cb53-6"><a href="chapter5.html#cb53-6"></a>        <span class="dt">OrgDb =</span> org.Hs.eg.db)</span>
<span id="cb53-7"><a href="chapter5.html#cb53-7"></a><span class="kw">head</span>(gene.df)</span></code></pre></div>
<pre><code>##   ENTREZID         ENSEMBL SYMBOL
## 1     4312 ENSG00000196611   MMP1
## 2     8318 ENSG00000093009  CDC45
## 3    10874 ENSG00000109255    NMU
## 4    55143 ENSG00000134690  CDCA8
## 5    55388 ENSG00000065328  MCM10
## 6      991 ENSG00000117399  CDC20</code></pre>
<div class="sourceCode" id="cb55"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb55-1"><a href="chapter5.html#cb55-1"></a>ggo &lt;-<span class="st"> </span><span class="kw">groupGO</span>(<span class="dt">gene     =</span> gene,</span>
<span id="cb55-2"><a href="chapter5.html#cb55-2"></a>               <span class="dt">OrgDb    =</span> org.Hs.eg.db,</span>
<span id="cb55-3"><a href="chapter5.html#cb55-3"></a>               <span class="dt">ont      =</span> <span class="st">&quot;CC&quot;</span>,</span>
<span id="cb55-4"><a href="chapter5.html#cb55-4"></a>               <span class="dt">level    =</span> <span class="dv">3</span>,</span>
<span id="cb55-5"><a href="chapter5.html#cb55-5"></a>               <span class="dt">readable =</span> <span class="ot">TRUE</span>)</span>
<span id="cb55-6"><a href="chapter5.html#cb55-6"></a></span>
<span id="cb55-7"><a href="chapter5.html#cb55-7"></a><span class="kw">head</span>(ggo)</span></code></pre></div>
<pre><code>##                    ID                    Description Count
## GO:0005886 GO:0005886                plasma membrane    55
## GO:0005628 GO:0005628              prospore membrane     0
## GO:0005789 GO:0005789 endoplasmic reticulum membrane     8
## GO:0019867 GO:0019867                 outer membrane     3
## GO:0031090 GO:0031090             organelle membrane    16
## GO:0034357 GO:0034357        photosynthetic membrane     0
##            GeneRatio
## GO:0005886    55/207
## GO:0005628     0/207
## GO:0005789     8/207
## GO:0019867     3/207
## GO:0031090    16/207
## GO:0034357     0/207
##                                                                                                                                                                                                                                                                                                                                                      geneID
## GO:0005886 S100A9/MELK/S100A8/MARCO/ASPM/CXCL10/LAMP3/CEP55/UGT8/UBE2C/SLC7A5/CXCL9/FADS2/MSLN/IL1R2/KIF18A/S100P/GZMB/TRAT1/GABRP/AQP9/GPR19/SLC2A6/KIF20A/LAG3/NUDT1/CACNA1D/VSTM4/ITPR1/SYT17/SLC16A4/CORIN/KCNK15/CA12/KCNE4/HLA-DQA1/ADH1B/PDZK1/C7/ACKR1/COL17A1/PSD3/EMCN/SLC44A4/LRP2/NLGN4X/MAPT/ERBB4/CX3CR1/LAMP5/ABCA8/STEAP4/PTPRT/TMC5/CYBRD1
## GO:0005628                                                                                                                                                                                                                                                                                                                                                 
## GO:0005789                                                                                                                                                                                                                                                                                               FADS2/CDK1/CHODL/ITPR1/HLA-DQA1/CYP4F8/CYP4B1/FMO5
## GO:0019867                                                                                                                                                                                                                                                                                                                                  BCL2A1/MAOB/PGR
## GO:0031090                                                                                                                                                                                                                                                MARCO/BCL2A1/LAMP3/DUSP2/SLC2A6/DTL/NUDT1/MAOB/ITPR1/GASK1B/HLA-DQA1/LRP2/LAMP5/STEAP4/PGR/CYBRD1
## GO:0034357</code></pre>
<p>The input parameters of <em>gene</em> is a vector of gene IDs (can be any ID type that supported by corresponding <code>OrgDb</code>).</p>
<p>If <em>readable</em> is setting to <em>TRUE</em>, the input gene IDs will be converted to gene symbols.</p>
</div>
<div id="go-over-representation-test" class="section level2">
<h2><span class="header-section-number">5.3</span> GO over-representation test</h2>
<p>Over-representation test<span class="citation">(Boyle et al. <a href="#ref-boyle2004" role="doc-biblioref">2004</a>)</span> were implemented in <a href="https://www.bioconductor.org/packages/clusterProfiler">clusterProfiler</a>. For calculation details and explanation of paramters, please refer to the vignette of <a href="https://www.bioconductor.org/packages/DOSE">DOSE</a>.</p>
<div class="sourceCode" id="cb57"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb57-1"><a href="chapter5.html#cb57-1"></a>ego &lt;-<span class="st"> </span><span class="kw">enrichGO</span>(<span class="dt">gene          =</span> gene,</span>
<span id="cb57-2"><a href="chapter5.html#cb57-2"></a>                <span class="dt">universe      =</span> <span class="kw">names</span>(geneList),</span>
<span id="cb57-3"><a href="chapter5.html#cb57-3"></a>                <span class="dt">OrgDb         =</span> org.Hs.eg.db,</span>
<span id="cb57-4"><a href="chapter5.html#cb57-4"></a>                <span class="dt">ont           =</span> <span class="st">&quot;CC&quot;</span>,</span>
<span id="cb57-5"><a href="chapter5.html#cb57-5"></a>                <span class="dt">pAdjustMethod =</span> <span class="st">&quot;BH&quot;</span>,</span>
<span id="cb57-6"><a href="chapter5.html#cb57-6"></a>                <span class="dt">pvalueCutoff  =</span> <span class="fl">0.01</span>,</span>
<span id="cb57-7"><a href="chapter5.html#cb57-7"></a>                <span class="dt">qvalueCutoff  =</span> <span class="fl">0.05</span>,</span>
<span id="cb57-8"><a href="chapter5.html#cb57-8"></a>        <span class="dt">readable      =</span> <span class="ot">TRUE</span>)</span>
<span id="cb57-9"><a href="chapter5.html#cb57-9"></a><span class="kw">head</span>(ego)</span></code></pre></div>
<pre><code>##                    ID
## GO:0005819 GO:0005819
## GO:0005876 GO:0005876
## GO:0000779 GO:0000779
## GO:0072686 GO:0072686
## GO:0000775 GO:0000775
## GO:0000776 GO:0000776
##                                         Description
## GO:0005819                                  spindle
## GO:0005876                      spindle microtubule
## GO:0000779 condensed chromosome, centromeric region
## GO:0072686                          mitotic spindle
## GO:0000775           chromosome, centromeric region
## GO:0000776                              kinetochore
##            GeneRatio   BgRatio       pvalue     p.adjust
## GO:0005819    25/200 272/11816 4.695505e-12 1.493171e-09
## GO:0005876    12/200  48/11816 1.623758e-11 2.581776e-09
## GO:0000779    15/200  91/11816 2.808998e-11 2.628919e-09
## GO:0072686    15/200  92/11816 3.306816e-11 2.628919e-09
## GO:0000775    18/200 154/11816 1.061302e-10 6.749884e-09
## GO:0000776    15/200 107/11816 3.047081e-10 1.614953e-08
##                  qvalue
## GO:0005819 1.378996e-09
## GO:0005876 2.384361e-09
## GO:0000779 2.427899e-09
## GO:0072686 2.427899e-09
## GO:0000775 6.233756e-09
## GO:0000776 1.491466e-08
##                                                                                                                                                         geneID
## GO:0005819 CDCA8/CDC20/KIF23/CENPE/ASPM/DLGAP5/SKA1/NUSAP1/TPX2/TACC3/NEK2/CDK1/MAD2L1/KIF18A/BIRC5/KIF11/TTK/AURKB/PRC1/KIFC1/KIF18B/KIF20A/AURKA/CCNB1/KIF4A
## GO:0005876                                                                             CENPE/SKA1/NUSAP1/CDK1/KIF18A/BIRC5/KIF11/AURKB/PRC1/KIF18B/AURKA/KIF4A
## GO:0000779                                                            CENPE/NDC80/HJURP/SKA1/NEK2/CENPM/CENPN/ERCC6L/MAD2L1/CDT1/BIRC5/NCAPG/AURKB/AURKA/CCNB1
## GO:0072686                                                           KIF23/CENPE/ASPM/SKA1/NUSAP1/TPX2/TACC3/CDK1/MAD2L1/KIF18A/KIF11/AURKB/KIFC1/KIF18B/AURKA
## GO:0000775                                           CDCA8/CENPE/NDC80/HJURP/SKA1/NEK2/CENPM/CENPN/ERCC6L/MAD2L1/KIF18A/CDT1/BIRC5/TTK/NCAPG/AURKB/AURKA/CCNB1
## GO:0000776                                                             CENPE/NDC80/HJURP/SKA1/NEK2/CENPM/CENPN/ERCC6L/MAD2L1/KIF18A/CDT1/BIRC5/TTK/AURKB/CCNB1
##            Count
## GO:0005819    25
## GO:0005876    12
## GO:0000779    15
## GO:0072686    15
## GO:0000775    18
## GO:0000776    15</code></pre>
<p>As I mentioned before, any gene ID type that supported in <code>OrgDb</code> can be directly used in GO analyses. User need to specify the <code>keyType</code> parameter to specify the input gene ID type.</p>
<div class="sourceCode" id="cb59"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb59-1"><a href="chapter5.html#cb59-1"></a>ego2 &lt;-<span class="st"> </span><span class="kw">enrichGO</span>(<span class="dt">gene         =</span> gene.df<span class="op">$</span>ENSEMBL,</span>
<span id="cb59-2"><a href="chapter5.html#cb59-2"></a>                <span class="dt">OrgDb         =</span> org.Hs.eg.db,</span>
<span id="cb59-3"><a href="chapter5.html#cb59-3"></a>                <span class="dt">keyType       =</span> <span class="st">&#39;ENSEMBL&#39;</span>,</span>
<span id="cb59-4"><a href="chapter5.html#cb59-4"></a>                <span class="dt">ont           =</span> <span class="st">&quot;CC&quot;</span>,</span>
<span id="cb59-5"><a href="chapter5.html#cb59-5"></a>                <span class="dt">pAdjustMethod =</span> <span class="st">&quot;BH&quot;</span>,</span>
<span id="cb59-6"><a href="chapter5.html#cb59-6"></a>                <span class="dt">pvalueCutoff  =</span> <span class="fl">0.01</span>,</span>
<span id="cb59-7"><a href="chapter5.html#cb59-7"></a>                <span class="dt">qvalueCutoff  =</span> <span class="fl">0.05</span>)</span></code></pre></div>
<p>Gene ID can be mapped to gene Symbol by using paramter <code>readable=TRUE</code> or <code>setReadable</code> function.</p>
<div class="sourceCode" id="cb60"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb60-1"><a href="chapter5.html#cb60-1"></a>ego2 &lt;-<span class="st"> </span><span class="kw">setReadable</span>(ego2, <span class="dt">OrgDb =</span> org.Hs.eg.db)</span></code></pre></div>
<div id="drop-specific-go-terms-or-level" class="section level3">
<h3><span class="header-section-number">5.3.1</span> drop specific GO terms or level</h3>
<p><code>enrichGO</code> test the whole GO corpus and enriched result may contains very general terms. With <code>dropGO</code> function, user can remove specific GO terms or GO level from results obtained from both <code>enrichGO</code> and <code>compareCluster</code>.</p>
</div>
<div id="test-go-at-sepcific-level" class="section level3">
<h3><span class="header-section-number">5.3.2</span> test GO at sepcific level</h3>
<p><code>enrichGO</code> doesn’t contain parameter to restrict the test at specific GO level. Instead, we provide a function <code>gofilter</code> to restrict the result at specific GO level. It works with results obtained from both <code>enrichGO</code> and <code>compareCluster</code>.</p>
</div>
<div id="reduce-redundancy-of-enriched-go-terms" class="section level3">
<h3><span class="header-section-number">5.3.3</span> reduce redundancy of enriched GO terms</h3>
<p>GO is organized in parent-child structure, thus a parent term can be overlap with a large proportion with all its child terms. This can result in redundant findings. To solve this issue, <a href="http://bioconductor.org/packages/clusterProfiler">clusterProfiler</a> implement <a href="https://github.com/GuangchuangYu/clusterProfiler/issues/28"><code>simplify</code></a> method to reduce redundant GO terms from the outputs of <code>enrichGO</code> and <code>gseGO</code>. The function internally called <a href="http://bioconductor.org/packages/GOSemSim">GOSemSim</a> <span class="citation">(Yu et al. <a href="#ref-yu2010" role="doc-biblioref">2010</a>)</span> to calculate semantic similarity among GO terms and remove those highly similar terms by keeping one representative term. An example can be found in <a href="https://guangchuangyu.github.io/2015/10/use-simplify-to-remove-redundancy-of-enriched-go-terms/">the blog post</a>.</p>
</div>
</div>
<div id="go-gene-set-enrichment-analysis" class="section level2">
<h2><span class="header-section-number">5.4</span> GO Gene Set Enrichment Analysis</h2>
<p>A common approach in analyzing gene expression profiles was identifying differential expressed genes that are deemed interesting. The enrichment analysis we demonstrated previous were based on these differential expressed genes. This approach will find genes where the difference is large, but it will not detect a situation where the difference is small, but evidenced in coordinated way in a set of related genes. Gene Set Enrichment Analysis (GSEA)<span class="citation">(Subramanian et al. <a href="#ref-subramanian_gene_2005" role="doc-biblioref">2005</a>)</span> directly addresses this limitation. All genes can be used in GSEA; GSEA aggregates the per gene statistics across genes within a gene set, therefore making it possible to detect situations where all genes in a predefined set change in a small but coordinated way. Since it is likely that many relevant phenotypic differences are manifested by small but consistent changes in a set of genes.</p>
<p>For algorithm details, please refer to the vignette of <a href="https://www.bioconductor.org/packages/DOSE">DOSE</a>.</p>
<div class="sourceCode" id="cb61"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb61-1"><a href="chapter5.html#cb61-1"></a>ego3 &lt;-<span class="st"> </span><span class="kw">gseGO</span>(<span class="dt">geneList     =</span> geneList,</span>
<span id="cb61-2"><a href="chapter5.html#cb61-2"></a>              <span class="dt">OrgDb        =</span> org.Hs.eg.db,</span>
<span id="cb61-3"><a href="chapter5.html#cb61-3"></a>              <span class="dt">ont          =</span> <span class="st">&quot;CC&quot;</span>,</span>
<span id="cb61-4"><a href="chapter5.html#cb61-4"></a>              <span class="dt">nPerm        =</span> <span class="dv">1000</span>,</span>
<span id="cb61-5"><a href="chapter5.html#cb61-5"></a>              <span class="dt">minGSSize    =</span> <span class="dv">100</span>,</span>
<span id="cb61-6"><a href="chapter5.html#cb61-6"></a>              <span class="dt">maxGSSize    =</span> <span class="dv">500</span>,</span>
<span id="cb61-7"><a href="chapter5.html#cb61-7"></a>              <span class="dt">pvalueCutoff =</span> <span class="fl">0.05</span>,</span>
<span id="cb61-8"><a href="chapter5.html#cb61-8"></a>              <span class="dt">verbose      =</span> <span class="ot">FALSE</span>)</span></code></pre></div>
<p>GSEA use permutation test, user can set <em>nPerm</em> for number of permutations. Only gene Set size in <code>[minGSSize, maxGSSize]</code> will be tested.</p>
<p>If you have issues in preparing your own <code>geneList</code>, please refer to the <a href="https://github.com/GuangchuangYu/DOSE/wiki/how-to-prepare-your-own-geneList">wiki
page</a>.</p>
</div>
<div id="go-semantic-similarity-analysis" class="section level2">
<h2><span class="header-section-number">5.5</span> GO Semantic Similarity Analysis</h2>
<p>GO semantic similarity can be calculated by <a href="https://www.bioconductor.org/packages/GOSemSim">GOSemSim</a><span class="citation">(Yu et al. <a href="#ref-yu2010" role="doc-biblioref">2010</a>)</span>. We can use it to cluster genes/proteins into different clusters based on their functional similarity and can also use it to measure the similarities among GO terms to reduce the redundancy of GO enrichment results.</p>
<div id="go-analysis-for-non-model-organisms" class="section level3">
<h3><span class="header-section-number">5.5.1</span> GO analysis for non-model organisms</h3>
<p>Both <code>enrichGO</code> and <code>gseGO</code> functions require an <code>OrgDb</code> object as background annotation. For organisms that don’t have <code>OrgDb</code> provided by Bioconductor, users can query one (if available) online via <a href="http://bioconductor.org/packages/AnnotationHub">AnnotationHub</a>. If there is no <code>OrgDb</code> available, users can obtain GO annotation from other sources, e.g. from <a href="http://bioconductor.org/packages/biomaRt">biomaRt</a> or <a href="https://www.blast2go.com/">Blast2GO</a>. Then using <code>enricher</code> or <code>GSEA</code> function to analyze, similar to the examples using wikiPathways and MSigDB. Another solution is to create <code>OrgDb</code> by your own using <a href="http://bioconductor.org/packages/AnnotationForge">AnnotationForge</a> package.</p>

</div>
</div>
</div>
<h3>References</h3>
<div id="refs" class="references">
<div id="ref-boyle2004">
<p>Boyle, Elizabeth I, Shuai Weng, Jeremy Gollub, Heng Jin, David Botstein, J Michael Cherry, and Gavin Sherlock. 2004. “GO::TermFinder–open Source Software for Accessing Gene Ontology Information and Finding Significantly Enriched Gene Ontology Terms Associated with a List of Genes.” <em>Bioinformatics (Oxford, England)</em> 20 (18): 3710–5. <a href="https://doi.org/10.1093/bioinformatics/bth456">https://doi.org/10.1093/bioinformatics/bth456</a>.</p>
</div>
<div id="ref-subramanian_gene_2005">
<p>Subramanian, Aravind, Pablo Tamayo, Vamsi K. Mootha, Sayan Mukherjee, Benjamin L. Ebert, Michael A. Gillette, Amanda Paulovich, et al. 2005. “Gene Set Enrichment Analysis: A Knowledge-Based Approach for Interpreting Genome-Wide Expression Profiles.” <em>Proceedings of the National Academy of Sciences of the United States of America</em> 102 (43): 15545–50. <a href="https://doi.org/10.1073/pnas.0506580102">https://doi.org/10.1073/pnas.0506580102</a>.</p>
</div>
<div id="ref-yu2010">
<p>Yu, Guangchuang, Fei Li, Yide Qin, Xiaochen Bo, Yibo Wu, and Shengqi Wang. 2010. “GOSemSim: An R Package for Measuring Semantic Similarity Among Go Terms and Gene Products.” <em>Bioinformatics</em> 26 (7): 976–78. <a href="https://doi.org/10.1093/bioinformatics/btq064">https://doi.org/10.1093/bioinformatics/btq064</a>.</p>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="chapter4.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="chapter6.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"google": false,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "google", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"download": null,
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
